### DISCUSSION TITLE
Using multiple GitHub accounts on one browser

### DISCUSSION THREAD

**Discussion by ankur-lt** [community/community#22014] [Enterprise] [answered]
<p>Hello,<br>
I am using 2 GitHub accounts, one by <strong>personal email</strong> for open source contributions and <strong>one of work email for contributing to the work repos</strong>. I have been added to the organization on work email.<br>
Thus, its a pain to logout work email and login personal for reviewing PRs in personal projects, and vice versa <strong>on the browser</strong>. Is there any way so that I can use both GitHub accounts on one browser or merge work account to personal one keeping the organization there with the account?</p>
<p>Thanks in advance!</p>

**Comment by canuckjacq** [comment]
<p>:wave: Welcome!</p>
<p>As you’ve discovered, GitHub really isn’t made for using with more than one account.</p>
<p>We don’t have a way to auto-magically merge accounts. We do have a document on how to move your work from one account to another:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/merging-multiple-user-accounts" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/merging-multiple-user-accounts" target="_blank" rel="noopener">Merging multiple user accounts - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/merging-multiple-user-accounts" target="_blank" rel="noopener">//docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/merging-multiple-user-accounts</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>In order to get your contributions attributed to the new account, you will have to make sure you add the email addresses you’ve been using for commits.</p>
<p>Things that don’t transfer (this list is not exhaustive):</p>
<ol>
<li>
<p>Access to organisations or repositories. If you have access to other people’s organisations or repositories on one account, you can’t move that access to a different account. You will have to ask the people who own the repository or the organisation admins to invite the account you wish to keep.</p>
</li>
<li>
<p>Student Developer Pack/other offers granting discounted or free Pro access. These remain on the account they were granted to and until the natural expiry of that offer, you won’t be able to reapply with a different account even if you move your email address.</p>
</li>
<li>
<p>SSH keys, sponsorships, account creation date, profile information</p>
</li>
</ol>
<p>I hope that helps</p>

**Comment by ankur-lt** [comment]
<p>Thanks, <a class="mention" href="/canuckjacq">@canuckjacq</a> for the help.<br>
I believe most developers use their work GitHub account using the work email. GitHub should add a feature to use multiple accounts in one browser, similar to Google accounts!</p>

**Reply by haddar340** [reply]
i agree. it's helpful

**Reply by yuezk** [reply]
I write an extension to support switching multiple GitHub accounts in the browser seamlessly. You may give it a try. It has been published to the [Chrome Web Store](https://chrome.google.com/webstore/detail/github-account-switcher/glnlddiaebecckddpfcangohjebhhoea) and [Firefox add-on](https://addons.mozilla.org/en-US/firefox/addon/github-account-switcher/). Here is the source code: https://github.com/yuezk/github-account-switcher

**Reply by ghost** [reply]
> I write an extension to support switching multiple GitHub accounts in the browser seamlessly. You may give it a try. It has been published to the [Chrome Web Store](https://chrome.google.com/webstore/detail/github-account-switcher/glnlddiaebecckddpfcangohjebhhoea) and [Firefox add-on](https://addons.mozilla.org/en-US/firefox/addon/github-account-switcher/). Here is the source code: https://github.com/yuezk/github-account-switcher

It works beautifully in Firefox! Thank you! :)

**Reply by TheColorRed** [reply]
> I write an extension to support switching multiple GitHub accounts in the browser seamlessly. You may give it a try. It has been published to the [Chrome Web Store](https://chrome.google.com/webstore/detail/github-account-switcher/glnlddiaebecckddpfcangohjebhhoea) and [Firefox add-on](https://addons.mozilla.org/en-US/firefox/addon/github-account-switcher/). Here is the source code: https://github.com/yuezk/github-account-switcher

This is working for me on chrome! I can switch between my personal and my github enterprise account.

**Reply by Abner97** [reply]
> I write an extension to support switching multiple GitHub accounts in the browser seamlessly. You may give it a try. It has been published to the [Chrome Web Store](https://chrome.google.com/webstore/detail/github-account-switcher/glnlddiaebecckddpfcangohjebhhoea) and [Firefox add-on](https://addons.mozilla.org/en-US/firefox/addon/github-account-switcher/). Here is the source code: https://github.com/yuezk/github-account-switcher

Works like a charm! Thank you man.

**Reply by atomlab** [reply]
> I write an extension to support switching multiple GitHub accounts in the browser seamlessly. You may give it a try. It has been published to the [Chrome Web Store](https://chrome.google.com/webstore/detail/github-account-switcher/glnlddiaebecckddpfcangohjebhhoea) and [Firefox add-on](https://addons.mozilla.org/en-US/firefox/addon/github-account-switcher/). Here is the source code: https://github.com/yuezk/github-account-switcher

This is what I needed. Thank you!

**Reply by migldasilva** [reply]
Actually, Microsoft already does it for Azure. You can be logged in with multiple accounts.

**Comment by airtower-luna** [comment]
<p>At least in Firefox you can use the “containers” feature to separate your accounts in one browser. :slightly_smiling_face:<br>
<a href="https://support.mozilla.org/en-US/kb/containers" class="onebox" target="_blank" rel="noopener">https://support.mozilla.org/en-US/kb/containers</a></p>

**Comment by ankur-lt** [comment]
<p>Woahh!!! This will be very useful!</p>

**Comment by devendra-rgb** [comment]
<p>Thanks for suggestion really it is very useful</p>

**Comment by 01005591-tp** [comment]
I have started using Vivaldi Browser recently and ended up with their [User profiles](https://help.vivaldi.com/desktop/tools/user-profiles/).

**Comment by kmaehashi** [comment]
We've been using Chrome browser profiles to switch between work/personal GitHub accounts for months, but the experience is poor. Specifically when clicking a GitHub link on other apps like Slack, the link will open in the most-recently-used browser profile, which is only sometimes an expected one.

As a workaround, we are using [Choosy](https://choosy.app/) or [Finicky](https://github.com/johnste/finicky) to switch the profile based on the URL being opened, but this requires each employee to maintain a long list of URL patterns like:

```json
        "github.com/mycorp-*",
        "github.com/*_mycorp",
        "*.github.com/mycorp-*",
        "*.githubusercontent.com/mycorp-*",
        "github.com/orgs/mycorp-*",
        "github.com/organizations/mycorp-*",
        "mycorp-*.github.io/*",
        "*.pages.github.io/*"
```

This is not even a complete list. In addition, AFAIK no solutions like this are available on Windows or Linux environments.

Now that [GitHub Mobile supports multiple accounts](https://github.blog/changelog/2023-08-15-log-in-with-multiple-github-accounts-on-github-mobile/), I think it would make sense to support it on the Web as well.

**Reply by yuezk** [reply]
Hi @kmaehashi, have you ever tried this browser extension, https://github.com/yuezk/github-account-switcher? Of course, the best is always the official support.

**Reply by kmaehashi** [reply]
Hi @yuezk! Actually, I’m aware of your extension since this April 😃

In my opinion, one of the core values GitHub Enterprise provides is enhanced security, including authentication through IdP or SSO enforcement. I never intend to offend you – I’m also a publisher of a Chrome extension – but giving third-party extensions control of session cookies, which is the core part of the authentication, is obviously unhealthy and was an unacceptable solution for us, unfortunately.

The existence of such an extension is proof of the importance of this feature. I'd expect GitHub to natively support multiple accounts on the Web.

**Reply by yuezk** [reply]
I fully agree with you. The extension is just a workaround for the requirements.

**Comment by billythekid** [comment]
This functionality recently rolled out to our Mobile app: https://github.blog/changelog/2023-08-15-log-in-with-multiple-github-accounts-on-github-mobile/

More support is on our roadmap: https://github.com/orgs/github/projects/4247/views/1?filterQuery=-status%3A%22Q4+2022+%E2%80%93+Oct-Dec%22+accoun&pane=issue&itemId=32913512

**Comment by hpsin** [comment, marked as answer]
Hey all!
[We shipped multi account support last week](https://github.blog/changelog/2023-11-03-multi-account-support-on-github-com/). It remembers account sessions in your browser so that you can switch between them without doing the entire sign in flow again.

Hopefully this helps with the hassle. Let us know if you have any feedback!

**Reply by ittaysw** [reply]
This is a great improvement. One feedback is that it would be nice to switch accounts automatically based on projects (business & private accounts). There's currently a Chrome extension that does it but it stops working every few days. So maybe an approach is to improve it or provide ways in which people can do it as a hack (e.g. greasemonkey script)

**Reply by XJDKC** [reply]
+1 here, it would be great to switch accounts automatically based on repos.

**Reply by mattbaker-digital** [reply]
When you navigate directly to an org with SSO while signed in on a personal account not associated with that org, you get the SSO dialog with a single continue button and no way from that page to switch accounts. You have to go back to a standard page to switch accounts and then navigate back there.
